### PR TITLE
ci: Add Dependency Review linting tool

### PR DESCRIPTION
**Pull request by reedloden**
Use https://github.com/actions/dependency-review-action to scan all PRs for out-of-date dependencies and for license enforcement.

https://docs.github.com/en/code-security/supply-chain-security/understanding-your-software-supply-chain/about-dependency-review#dependency-review-enforcement

### CODE REVIEW

**Comment by wadells** [comment]
Looking at https://github.com/gravitational/teleport/actions/runs/3107953419/jobs/5036625323#step:3:9

> We could not detect a license for the following dependencies:
>
> .github/workflows/dependency-review.yaml » gravitational/shared-workflows/.github/workflows/dependency-review.yaml@main

This seems like confusing output -- I'm not sure what to make of it.  This isn't a blocker, but it seems like the upstream could do with a more clear ux.

**Review by wadells — APPROVED** [review]

**Comment by reedloden** [comment]
> Looking at https://github.com/gravitational/teleport/actions/runs/3107953419/jobs/5036625323#step:3:9
> 
> > We could not detect a license for the following dependencies:
> > .github/workflows/dependency-review.yaml » gravitational/shared-workflows/.github/workflows/dependency-review.yaml@main
> 
> This seems like confusing output -- I'm not sure what to make of it. This isn't a blocker, but it seems like the upstream could do with a more clear ux.

That just means it couldn't find a license for that action (I guess I need to figure out how to add licenses to actions).

https://github.com/actions/dependency-review-action#considerations

> We don't have license information for all of your dependents. If we can't detect the license for a dependency we will inform you, but the action won't fail.

So, yeah, non-blocking issue.

File: .github/workflows/dependency-review.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Dependency Review
+
+on:
+  push:
+    branches:
+      - master
+      - branch/*
+  pull_request:
```
[the remark this answers is not in the document]
Comment (reedloden) [reply .github/workflows/dependency-review.yaml:8]: Yup, just means it'll apply to all PRs. Figured it was a tiny check and useful no matter if people are working on private branches.

**Review by zmb3 — APPROVED** [review]

**Comment by wadells** [comment]
> > We don't have license information for all of your dependents. If we can't detect the license for a dependency we will inform you, but the action won't fail.

This feels like "fail open" behavior to me.  In the case of dependency review, I think we probably want "fail closed" with some way to override. (e.g. a way we can force set a licence for a particular dependency).  Since we don't want "perfect" to get in the way of "better", I filed https://github.com/gravitational/SecOps/issues/215 for this.

**Pull request merged**